dev-docs/development-overview.md

```diff
@@ -5,6 +5,35 @@
 We're using Go's modules now. For an introduction, see:
 https://blog.golang.org/using-go-modules
 
+### How to add a new dependency
+
+When you want to use a new module in your code, just add an `import` statement for the module and use it:
+
+```go
+import "rsc.io/quote"
+. . .
+func Hello() string {
+    return quote.Hello()
+}
+```
+
+The next time you build, the module will be automatically downloaded and your go.mod/go.sum files updated.
+Be sure to commit the go.mod/go.sum changes.
+
+You can check that the imported module is in place with, for example:
+
+```shell
+go list -m rsc.io/q...
+```
+
+### How to remove a dependency
+
+It is important to be a good steward and clean up after yourself when you remove a dependency.
+Using `go build` can easily tell when something is missing and needs to be added, but not when it can safely be removed.
+The `go mod tidy` command is provided to do this extra work, but you must run it explicitly after you have removed uses of the no-longer-needed dependency from your code.
```

Comment:
My previous work left the `scripts/revendor` script in place, I think it’s the only “right” way to remove a dep at the moment.

There is a bit of a weird situation where we are still vendoring protos and still using `go get` to fetch the repos with those protos, but those repos don’t have any go files in them. The process of fetching the proto repos adds the modules to the `go.sum` file, but `go mod tidy` will remove them. So you have to revendor the protos or else there’s churn in the `go.sum`, which I *think* will fail the repo health check. The revendor script does all of these things in the desired order.

I’ll try to find a way to clean this up. Perhaps making the protovendor directory a separate go module would do it.

Reply:
@danielsdeleo Do you want to also update this PR, or put it on hold until you have a chance to do the work you mention?

Reply:
#4346 should clean things up

Reply:
Cool; it sounds like this doc file needs updating a bit due to #4346, though I am not clear on precisely what the changes should be. @danielsdeleo could you push a commit at least with the basics...?

Reply:
@msorens #4346 makes what you wrote here correct :) 

Longer explanation: prior to #4346, you needed to run `scripts/revendor.sh` to remove a go dependency because our protobuf and golang dependency management were intertwined, which meant we used go tools to fetch some repos with no go files in them. Fetching those repos with `go get` would add them to the the `.mod` and `.sum` files, but running `mod tidy` would remove them because they didn't have any go files (and thus nothing did/could import them). Now protobuf dependency management is contained to the `protovendor` part of the tree (which is its own go module now), so the standard go tooling can be used. That said, you can use the revendor script if you want.